opentelemetry/src/context.rs

```diff
@@ -897,4 +979,171 @@ mod tests {
         assert_eq!(Context::current().get::<ValueA>(), None);
         assert_eq!(Context::current().get::<ValueB>(), None);
     }
+
+    #[test]
+    fn test_is_telemetry_suppressed() {
+        // Default context has suppression disabled
+        let cx = Context::new();
+        assert!(!cx.is_telemetry_suppressed());
+
+        // With suppression enabled
+        let suppressed = cx.with_telemetry_suppressed();
+        assert!(suppressed.is_telemetry_suppressed());
+    }
+
+    #[test]
+    fn test_with_telemetry_suppressed() {
+        // Start with a normal context
+        let cx = Context::new();
+        assert!(!cx.is_telemetry_suppressed());
+
+        // Create a suppressed context
+        let suppressed = cx.with_telemetry_suppressed();
+
+        // Original should remain unchanged
+        assert!(!cx.is_telemetry_suppressed());
+
+        // New context should be suppressed
+        assert!(suppressed.is_telemetry_suppressed());
+
+        // Test with values to ensure they're preserved
+        let cx_with_value = cx.with_value(ValueA(42));
+        let suppressed_with_value = cx_with_value.with_telemetry_suppressed();
+
+        assert!(!cx_with_value.is_telemetry_suppressed());
+        assert!(suppressed_with_value.is_telemetry_suppressed());
+        assert_eq!(suppressed_with_value.get::<ValueA>(), Some(&ValueA(42)));
+    }
+
+    #[test]
+    fn test_enter_telemetry_suppressed_scope() {
+        // Ensure we start with a clean context
+        let _reset_guard = Context::new().attach();
+
+        // Default context should not be suppressed
+        assert!(!Context::is_current_telemetry_suppressed());
+
+        // Enter a suppressed scope
+        {
+            let _guard = Context::enter_telemetry_suppressed_scope();
+            assert!(Context::is_current_telemetry_suppressed());
+            assert!(Context::current().is_telemetry_suppressed());
+        }
+
+        // After guard is dropped, should be back to unsuppressed
+        assert!(!Context::is_current_telemetry_suppressed());
+        assert!(!Context::current().is_telemetry_suppressed());
+    }
+
+    #[test]
+    fn test_is_current_telemetry_suppressed() {
```

Comment:
It looks like this test is redundant. `test_nested_suppression_scopes` is already covering what this test covers.